### PR TITLE
CMake configure_file for versioning

### PR DESCRIPTION
**Pull request by FryDay**
Probably a better way to handle versioning as the version no longer needs to be hardcoded into the ui.

### CODE REVIEW

File: src/ui.c [hunk 1]
```diff
@@ -2,6 +2,7 @@
 #include "ui.h"
 
 #include "msemu.h"
+#include "config.h"
```
Comment (CaptainNic) [inline src/ui.c:5]: Isn't it best practice to have config.h be the first include?

Comment (FryDay) [reply src/ui.c:5]: I honestly wasn't sure, but that makes sense to me.

**Comment by kbembedded** [comment]
@FryDay @CaptainNic Made some changes to grab the version from git and also not print out version on cmdline on startup

It looks like this is more or less the paradigm used for getting git tag? Let me know if this is silly or if there are any other suggestions.

**Review by FryDay — COMMENTED** [review]
Looks good to me :+1:

**Review by CaptainNic — COMMENTED** [review]
Can't test at the moment, but version logic seems reasonable.

Don't wait on me 👍

**Pull request merged**